core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java

```diff
@@ -264,4 +225,124 @@ public void setConfig(final String storeName, final Options options, final Map<S
             options.optimizeFiltersForHits();
         }
     }
+
+    private ManagedKStreams createManagedKStreams(final StreamRegistryConfiguration configuration) {
+        Properties kstreamsProperties = new Properties();
+        configuration.getKafkaStreamsConfig().getKstreamsProperties().forEach(kstreamsProperties::put);
+        kstreamsProperties.put(ROCKSDB_CONFIG_SETTER_CLASS_CONFIG, CustomRocksDBConfig.class);
+        TopicsConfig topicsConfig = configuration.getTopicsConfig();
+
+        ManagedKStreams managedKStreams = new ManagedKStreams(kstreamsProperties, topicsConfig);
+        return managedKStreams;
+
+    }
+
+    private ManagedInfraManager createManagedInfraManager(final StreamRegistryConfiguration configuration) {
+        InfraManagerConfig infraManagerConfig = configuration.getInfraManagerConfig();
+        String infraManagerClassName = infraManagerConfig.getClassName();
+
+        if (infraManagerClassName != null && !infraManagerClassName.isEmpty()) {
```

Comment:
should we require an infra manager ?
(e.g. should there be a precondition check with the opposite of this?)

Reply:
Added the pre-condition to ensure that `infraManager` is null before the `ManagedInfraManager` is created.

Reply:
Ok got it. Maybe `null` check on `infraManagerClassName` as `InfraManagerConfig.className` is marked `@NotNull` ( [here](https://github.com/homeaway/stream-registry/blob/master/core/src/main/java/com/homeaway/streamplatform/streamregistry/configuration/InfraManagerConfig.java#L31-L36) )

I'll update with `isEmpty` Preconditions check.